openlibrary/templates/admin/sponsorship.html

```diff
@@ -0,0 +1,137 @@
+$def with (summary)
+
+$ _x = ctx.setdefault('bodyid', 'admin')
+$ _x = ctx.setdefault('usergroup', 'admin')
+
+$ statuses = {0: 'Needs purchasing', 1: 'Needs digitizing', 2: 'Needs republishing', 3: 'Complete'}
+
+<div id="contentHead">
+  $:render_template("admin/menu")
+  <h1>Book Sponsorship</h1>
+</div>
+
+<div id="contentBody">
+
+  <p>We've sponsored <strong>$len(summary['books'])</strong> books</p>
+
+  <table class="sponsor">
+    $for status in range(4):
+      <tr>
+        <th class="status--$status">$statuses[status]</th>
+        <td>$summary['statuses'].get(status, 0)</td>
+      </tr>
+  </table>
+
+  <hr>
+
+  <table class="sponsor">
+    <tr>
+      <th>Internet Archive Revenue</th>
+      <td class="earnings--profit">\$$('{:,.2f}'.format(summary['total_cost_cents'] / 100.))</td>
+      <th>Better World Books Revenue</th>
+      <td class="earnings--profit">\$$('{:,.2f}'.format(summary['book_cost_cents'] / 100.))</td>
+    </tr>
+    <tr>
+      <th>Book Costs</th>
+      <td>\$$('{:,.2f}'.format(summary['book_cost_cents'] / 100.))</td>
+      <th>Est. Book Costs</th>
+      <td>\$$('{:,.2f}'.format(summary['est_book_cost_cents'] / 100.))</td>
+      <th>Δ Book Costs</th>
+      <td class="earnings--$('loss' if summary['delta_book_cost_cents'] < 0 else 'profit')">\$$('{:,.2f}'.format(summary['delta_book_cost_cents'] / 100.))</td>
+    </tr>
+    <tr>
+      <th>Scanning Costs</th>
+      <td>\$$('{:,.2f}'.format(summary['scan_cost_cents'] / 100.))</td>
+      <th>Est. Scanning Costs</th>
+      <td>\$$('{:,.2f}'.format(summary['est_scan_cost_cents'] / 100.))</td>
+      <th>Δ Scanning Costs</th>
+      <td class="earnings--$('loss' if summary['delta_scan_cost_cents'] < 0 else 'profit')">\$$('{:,.2f}'.format(summary['delta_scan_cost_cents'] / 100.))</td>
+    </tr>
+    <tr>
+      <th>Profit/Loss</th>
+      $ profit_loss = (summary['total_cost_cents'] - summary['book_cost_cents'] - summary['scan_cost_cents'])
+      <td class="earnings--$('loss' if profit_loss < 0 else 'profit')">\$$(profit_loss / 100.)</td>
+    </tr>
+  </table>
+
+  <hr>
+
+  <table class="sponsor">
+    <tr>
+      <th>Total Unscanned Books</th>
+      <td>$summary['total_unscanned_books']</td>
+      $if ctx.user and ctx.user.is_admin():
+        $ isbns = ','.join([s['identifier'].replace('isbn_', '') for s in summary['books']])
+        <td><a href="https://8882k.csb.app/#isbns=$isbns">Print Slips</a></td>
+    </tr>
+    <tr>
+      <th>Average Scan Cost</th>
+      $ num_total_books = len(summary['books'])
+      $ num_scanned_books = num_total_books - summary['total_unscanned_books']
+      $ avg_scan_cost = summary['scan_cost_cents'] / num_scanned_books
```

Comment:
Put this in the `summary` function; I like how almost all the computations are done in the python, and the template is relatively "dumb"